mcf-browser/src/back-end/utils.js

```diff
@@ -0,0 +1,115 @@
+/**
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/**
+ * Module contains helper functions for api calls to data commons as well as
+ * helper functions for parsing file to create local knowledge graph.
+ */
+
+import {Node} from './graph.js';
+const API_ROOT = 'https://api.datacommons.org';
+
+/**
+ * Gets all property labels of the given dcid that are in the DC KG.
+ *
+ * @param {string} dcid The dcid of the node to find property labels for.
+ * @return {Object} An object containing both 'in' and 'out' property labels.
+ */
+async function getRemotePropertyLabels(dcid) {
+  const targetUrl = API_ROOT + '/node/property-labels?dcids=' + dcid;
+  return fetch(targetUrl)
+      .then((res) => res.json())
+      .then((data) => JSON.parse(data.payload)[dcid]);
+}
+
+/**
+ * Gets all property values containing the given dcid, property label, and
+ * direction.
+ *
+ * @param {string} dcid The dcid of the node to find property value for.
+ * @param {string} label The property label to query for.
+ * @param {boolean} isInverse Direction of property label, false indicates
+ *     an outgoing label, true is an incoming label.
+ * @return {Object} An object containing all found values matching the query.
+ */
+async function getRemotePropertyValues(dcid, label, isInverse) {
+  const direction = isInverse ? 'in' : 'out';
+  const targetUrl =
+      (API_ROOT + '/node/property-values?limit=500&dcids=' + dcid +
+       '&property=' + label + '&direction=' + direction);
+
+  return fetch(targetUrl)
+      .then((res) => res.json())
+      .then((data) => JSON.parse(data.payload)[dcid])
+      .then((triples) => isInverse ? triples.in : triples.out);
+}
+
+/**
+ * Parses an Object returned from the DC REST get_values API to create a Node
+ * object from the value's dcid or to return the string value that the object
+ * holds.
+ *
+ * @param {Object} valueObj An object returned from DC REST get_values API.
+ * @return {Node | string} The created Node if the value object has a dcid,
+ *     otherwise the string of the value.
+ */
+function getValueFromValueObj(valueObj) {
+  if (!('dcid' in valueObj || 'value' in valueObj)) {
+    throw new Error('ERROR remote fetch no dcid or value prop: ' + valueObj);
```

Comment:
Just to confirm, is "dcid" or "value" property (as in, DC property)? Or are they fields in the object?

Reply:
They are fields in the object returned from the DC Rest API. If the target points to another node, then the returned object has a 'dcid' field. Otherwise, if the target is just a text based property/single value, then the returned object has a 'value' field. 
Should I make this more clear in the function docs and/or the error message?

Reply:
Got it. I was puzzled by the use of "prop" since it has a meaning wrt DC schema. Perhaps a slight tweak to the error message might help:

  'ERROR: DC API returned an object with no "dcid" or "value" field: ' + valueObj